test/jdk/com/sun/net/httpserver/simpleserver/FileServerHandlerTest.java

```diff
@@ -82,6 +84,82 @@ public void testNotImplementedRequestMethod(String requestMethod) throws Excepti
         assertEquals(exchange.rCode, 501);
     }
 
+    @DataProvider
+    public Object[][] validRangeHeaders() {
+        return new Object[][]{
+                // start-end
+                {1000L, "bytes=0-499", new FileServerHandler.RangeEntry(0, 499)},
+                {1000L, "bytes=500-999", new FileServerHandler.RangeEntry(500, 999)},
+                {500L,  "bytes=100-199", new FileServerHandler.RangeEntry(100, 199)},
+
+                // start-
+                {1000L, "bytes=0-", new FileServerHandler.RangeEntry(0, 999)},
+                {1000L, "bytes=500-", new FileServerHandler.RangeEntry(500, 999)},
+                {500L,  "bytes=250-", new FileServerHandler.RangeEntry(250, 499)},
+
+                // -length
+                {1000L, "bytes=-500", new FileServerHandler.RangeEntry(500, 999)},
+                {1000L, "bytes=-1", new FileServerHandler.RangeEntry(999, 999)},
+                {500L,  "bytes=-50", new FileServerHandler.RangeEntry(450, 499)},
+
+                // single byte ranges
+                {1000L, "bytes=0-0", new FileServerHandler.RangeEntry(0, 0)},
+                {1000L, "bytes=999-999", new FileServerHandler.RangeEntry(999, 999)},
+                {500L,  "bytes=499-499", new FileServerHandler.RangeEntry(499, 499)},
+
+                // multiple ranges
+                {
+                        1000L, "bytes=0-499,500-999",
+                        new FileServerHandler.RangeEntry(0, 499),
+                        new FileServerHandler.RangeEntry(500, 999)
+                },
+                {
+                        1000L, "bytes=0-,-200",
+                        new FileServerHandler.RangeEntry(0, 999),
+                        new FileServerHandler.RangeEntry(800, 999)
+                },
+                {
+                        500L, "bytes=0-99,200-",
+                        new FileServerHandler.RangeEntry(0, 99),
+                        new FileServerHandler.RangeEntry(200, 499)
+                }
+        };
+    }
+
+    @Test(dataProvider = "validRangeHeaders")
+    public void testValidRangeParse(long fileLength, String rangeHeader, FileServerHandler.RangeEntry... entries) {
+        var ranges = FileServerHandler.parseRangeHeader(rangeHeader, fileLength);
+        assertNotNull(ranges);
+        assertEquals(ranges.size(), entries.length);
+        for (int i = 0; i < entries.length; i++) {
+            assertEquals(ranges.get(i).start(), entries[i].start());
+            assertEquals(ranges.get(i).end(), entries[i].end());
+        }
+    }
+
+    @DataProvider
+    public Object[][] invalidRangeHeaders() {
+        return new Object[][]{
+                // invalid ranges: start > end, end beyond file, start beyond file
+                {"bytes=500-400"}, {"bytes=1000-1001"}, {"bytes=1000-"},
+
+                // malformed numbers or format
+                {"bytes=-0"}, {"bytes=meow"}, {"bytes=--500"}, {"bytes=500"}, {"bytes=-"}, {"bytes=+500-600"},
```

Comment:
Please add a test for `bytes=400--500` which uses a code path not covered by existing tests.
Also worth adding `bytes=500-+600` and variations with `+` in all possible positions, and omitting left part.
Add also `bytes=--`, `bytes=-+1`, ... I'm sure one could come up with more.